tasks/main.yml

```diff
@@ -60,16 +60,29 @@
   become: true
   changed_when: false
 
+- name: Look for grub configuration files
+  stat:
+   path: "{{ item }}"
+  register: grub_search
+  become: true
+  loop: "{{ grub_configuration_files }}"
+
 - name: replace current grub config
   copy:
     backup: "{{ backup_grub_config_file }}"
     # /etc/grub2.cfg -> ../boot/grub2/grub.cfg
     follow: true
     content: "{{ grub_mkconfig_result.stdout }}"
-    dest: "{{ item }}"
+    dest: "{{ item.stat.path }}"
+  when: item.stat.exists
   become: true
   register: grub_update_result
-  with_items:
-    - /etc/grub2.cfg
-    - /etc/grub2-efi.cfg
+  loop: "{{ grub_search.results }}"
   notify: "{{ kernel_restart_handler }}"
+
+- name: Update /boot/loader/entries with grubby
+  # NOTE: This is the new fangled way: https://unix.stackexchange.com/questions/557841/what-is-the-correct-way-to-handle-default-kernelopts-in-fedora-31
```

Comment:
Assume it works on older distros too?

Reply:
Good question. I've got a feeling this is redhat only. Think it is this tool: https://src.fedoraproject.org/rpms/grubby. Will see if ubuntu package it. Just merged the ubuntu support, so perhaps to make this redhat specific.